CONTRIBUTING.md

```diff
@@ -0,0 +1,209 @@
+
+If you're already familiar with our workflow, maybe have a quick look at the [pre-commit checks](#pre-commit-checks) directly below.
+
+## Pre-commit checks
+
+Before you commit any code, please perform the following checks:
+
+- [All tests pass](#testing): `$ nosetests beep`
+- [No style issues](#coding-style-guidelines): `$ flake8`
+
+## Workflow
+
+We use [GitHub](https://en.wikipedia.org/wiki/GitHub) standard [Fork and Pull Request workflow](https://gist.github.com/Chaser324/ce0505fbed06b947d962) to coordinate our work. 
+When making any kind of update, we try to follow the procedure below.
+
+### A. Before you begin
+
+1. Create an [issue](https://guides.github.com/features/issues/) where new proposals can be discussed before any coding is done.
+2. Create a personal [fork](https://help.github.com/articles/fork-a-repo/) the master repo.
+3. Download the source code onto your local system, by [cloning](https://help.github.com/articles/cloning-a-repository/) the repository (or your fork of the repository).
+4. Create a [branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/) of this repo on your own fork where all changes will be made
+5. [Install](#installation) BEEP with the developer options.
+6. Test if your installation worked. `nosetests beep`.
+
+You now have everything you need to start making changes!
+
+### B. Writing your code
+
+7. BEEP is developed in [Python](https://en.wikipedia.org/wiki/Python_(programming_language). Make sure to follow our [coding style guidelines](#coding-style-guidelines).
+8. Commit your changes to your branch with [useful, descriptive commit messages](https://chris.beams.io/posts/git-commit/). While developing, you can keep using the GitHub issue you're working on as a place for discussion. [Refer to your commits](https://stackoverflow.com/questions/8910271/how-can-i-reference-a-commit-in-an-issue-comment-on-github) when discussing specific lines of code.
+9. If you want to add a dependency on another library, or re-use code you found somewhere else, have a look at [these guidelines](#dependencies-and-reusing-code).
+
+### C. Merging your changes with BEEP
+
+9. [Test your code!](#testing)
+11. If you added a major new feature, perhaps it should be showcased in an [example notebook](#example-notebooks).
+12. When you feel your code is finished, or at least warrants serious discussion, run the [pre-commit checks](#pre-commit-checks) and then create a [pull request](https://help.github.com/articles/about-pull-requests/).
+13. Once a PR has been created, it will be reviewed by the maintainers. Changes might be suggested which you can make by simply adding new commits to the branch. When everything's finished, someone with the right GitHub permissions will merge your changes into BEEP repository.
+
+
+## Installation
+
+To install BEEP with all developer options, type:
+
+```bash
+pip install -e .[dev,docs]
```

Comment:
don't think we have the docs option, also our dev option is `tests` I think, e.g. `pip install -e .[tests]`

Reply:
Yeah, I forgot to update this. Fixed now.

Reply:
@chirranjeevigopal-TRI Did we want to make this `pip install -e .[tests]`?